### PR TITLE
ci: Project setup - CI config, licence/lint check and unit test (build)

### PR DESCRIPTION
**Pull request by rolsonquadras**
- gitignore file
- Circle CI config file
- Scripts for licence, lint(golangci-lint) and unit test along with Makefile
- Placeholder go file along with go.mod and go.sum file

Signed-off-by: Rolson Quadras <rolson.quadras@securekey.com>

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Binaries for programs and plugins
```
Comment (troyronda) [inline .gitignore]: Should have a license header.

Comment (rolsonquadras) [reply .gitignore]: Added - https://github.com/hyperledger/aries-framework-go/compare/0004b9425b06eddb52484f8fbfd11ffcc0d126e7..ffab3ad151e63cb395efa46935e28bfa1849f589

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+GO111MODULE=on
+
+all: checks unit-test
+
+checks: license lint
+
+lint:
+	@scripts/check_lint.sh
+
+license:
```
Comment (troyronda) [inline Makefile]: PHONY tag is missing for several targets.

Comment (rolsonquadras) [reply Makefile]: Added PHONY tag - https://github.com/hyperledger/aries-framework-go/compare/ffab3ad151e63cb395efa46935e28bfa1849f589..6a349ceee3704e5d4a1d8da0a776e7e4d5b83c5f

File: scripts/check_lint.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+set -e
+
+DOCKER_CMD=${DOCKER_CMD:-docker}
+
+if [ ! $(command -v ${DOCKER_CMD}) ]; then
+    exit 0
+fi
+
+echo "GolangCI Linter :: Started"
```
Comment (troyronda) [inline scripts/check_lint.sh]: check_license uses this format:
echo "Running check_license.sh"

(similarly for check_unit.sh)

Comment (rolsonquadras) [reply scripts/check_lint.sh]: Updated https://github.com/hyperledger/aries-framework-go/compare/6a349ceee3704e5d4a1d8da0a776e7e4d5b83c5f..56569012d7114fec0ec68e653956803876587e8c

**Pull request merged**